contributing: add markdown lints (#173)
rust-unofficial/patterns: CONTRIBUTING.md

Please make `Draft Pull requests` early so we can follow your progress and can give early feedback (see the following section).

## Check the article locally

Before submitting the PR launch the commands `mdbook build` to make sure that the book builds and `mdbook test` to make sure that
code examples are correct.

### Markdown lint

> To make sure the files comply with our Markdown style we use [markdownlint-cli](https://github.com/igorshubovych/markdownlint-cli). To spare you some manual work to get through the CI test you can use the following commands to automatically fix most of the emerging problems when writing Markdown files.

Reviewer: ```suggestion
To make sure the files comply with our Markdown style we use [markdownlint-cli](https://github.com/igorshubovych/markdownlint-cli). 
To spare you some manual work to get through the CI test you can use the following commands to automatically fix most of the emerging problems when writing Markdown files.
```

Please make `Draft Pull requests` early so we can follow your progress and can give early feedback (see the following section).

## Check the article locally

Before submitting the PR launch the commands `mdbook build` to make sure that the book builds and `mdbook test` to make sure that
code examples are correct.

### Markdown lint

> To make sure the files comply with our Markdown style we use [markdownlint-cli](https://github.com/igorshubovych/markdownlint-cli). 

Reviewer: ```suggestion
To make sure the files comply with our Markdown style we use [markdownlint-cli](https://github.com/igorshubovych/markdownlint-cli).
```

A good principle: "Work together, share ideas, teach others."

## Test the book locally before submitting

Before submitting the PR launch the commands `mdbook build` to make sure that the book builds and `mdbook test` to make sure that
code examples are correct.

### Markdown lint

> We use [markdownlint-cli](https://github.com/igorshubovych/markdownlint-cli) to check we write correct markdown files.

Reviewer: ```suggestion
To make sure the files comply with our Markdown style we use [markdownlint-cli](https://github.com/igorshubovych/markdownlint-cli). To spare you some manual work to get through the CI test you can use the following commands to automatically fix most of the emerging problems when writing Markdown files.
```

- Install:

  ```sh
  npm install -g markdownlint-cli
  ```

- Check all markdown files:
  - unix: `markdownlint '**/*.md'`
  - windows: `markdownlint **/*.md`

- Automatically fix basic errors:
  - unix: `markdownlint -f '**/*.md'`
  - windows: `markdownlint -f **/*.md`

> ### Important Note

Reviewer: Can we move up the `important note` to the PR section then?

Author: I moved all this "test" section away from the "Creating a PR" section in commit 0182425 because I think it wasn't that related.
In this way the contributing guidelines have a sort of "chronological order"

Reviewer: i meant important note is `###` which is under `Check you the article locally` put important note is regarding PRs

Author: Yes, now it should be under "Creating a Pull Request"
